opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -12,7 +12,214 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
+from collections import OrderedDict
+from collections import deque
+from collections import namedtuple
+import threading
+import typing
+
 from opentelemetry import trace as trace_api
+from opentelemetry.sdk import util
+
+try:
+    from collections.abc import MutableMapping
+    from collections.abc import Sequence
+except ImportError:
+    from collections import MutableMapping
+    from collections import Sequence
+
+
+MAX_NUM_ATTRIBUTES = 32
+MAX_NUM_ANNOTATIONS = 32
+MAX_NUM_EVENTS = 128
+MAX_NUM_LINKS = 32
+
+AttributeValue = typing.Union[str, bool, float]
+
+
+class BoundedList(Sequence):
```

Comment:
This class is likely to be used in metrics/logs as well, we might need to extract it to a common place. Probably not in this PR though.